### PR TITLE
Set up iOS for chat demo

### PR DESCRIPTION
**Pull request by hieuwu**
## What kind of change does this PR introduce?
Set up iOS projet for chat demo app
To resolve https://github.com/supabase-community/supabase-kt/issues/290
## What is the current behavior?
iOS version is not available

## What is the new behavior?
Chat Demo iOS version is available
Screenshot 2023-12-27 at 15 04 39

## Additional context
How to set up with XCode 15 and run the app on iOS 17
Follow this tutorial
[Make your cros-platform work on iOS](
https://www.jetbrains.com/help/kotlin-multiplatform-dev/multiplatform-integrate-in-existing-app.html#make-your-cross-platform-application-work-on-ios)

However, for **Framwork Search Paths**, the value should be `$(SRCROOT)/../common/build/xcode-frameworks/$(CONFIGURATION)/$(SDK_NAME)` , **Other linker flags** is `"$(inherited) -framework common"`

**Note**:
- First build the whole project in Android Studio
- If there's no iOS build config available in Android, open XCode then press build button

### CODE REVIEW

File: demos/chat-demo-mpp/gradle/libs.versions.toml [hunk 1]
```diff
@@ -1,16 +1,16 @@
...
-ktor = "2.3.1"
+koin = "3.4.3"
+ktor = "2.3.7"
 
 androidx-core = "1.10.1"
 androidx-compat = "1.6.1"
 androidx-lifecycle = "2.6.1"
 androidx-activity = "1.7.2"
-
+org-jetbrains-kotlin-jvm = "1.9.10"
```
Comment (jan-tennert) [inline demos/chat-demo-mpp/gradle/libs.versions.toml:13]: Two things:
- We can use the newest possible versions if we are updating them anyways (Kotlin 1.9.21, Compose 1.5.11)
- Is there a reason kotlin-jvm has a separate version entry?

**Review by jan-tennert — APPROVED** [review]
Other than that, looks good thanks! We might need to do some cleaning up in the common code (because of Compose and supabase-kt changes), but that is for another PR.

File: demos/chat-demo-mpp/gradle/libs.versions.toml (hunk 1 above)
Comment (hieuwu) [reply demos/chat-demo-mpp/gradle/libs.versions.toml:13]: - Let me try the latest posible versions
- No it's not. It should be the same version with Kotlin. I'll update it

Comment (hieuwu) [reply demos/chat-demo-mpp/gradle/libs.versions.toml]: Just updated. All good now

**Comment by hieuwu** [comment]
> Other than that, looks good thanks! We might need to do some cleaning up in the common code (because of Compose and supabase-kt changes), but that is for another PR.

Yeah let me make the demos work then go back to update the change accordingly

**Review by jan-tennert — APPROVED** [review]

**Pull request merged**